### PR TITLE
Refactoring load models

### PR DESCRIPTION
**Pull request by danielsalles**
In my tests there was no breach of compatibility. Any questions just comment :).

### CODE REVIEW

File: lib/assets/models/index.js [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
+const basename  = path.basename(module.filename);
+const env       = process.env.NODE_ENV || 'development';
+const config    = require(<%= configFile %>)[env];
 
-if (config.use_env_variable) {
-  var sequelize = new Sequelize(process.env[config.use_env_variable]);
-} else {
-  var sequelize = new Sequelize(config.database, config.username, config.password, config);
-}
+const starting = setting => setting.use_env_variable ?
```
Comment (Americas) [inline lib/assets/models/index.js]: Rename to `loadSequelize`?

File: lib/assets/models/index.js [hunk 2]
```diff
@@ -1,36 +1,34 @@
...
-  .readdirSync(__dirname)
-  .filter(function(file) {
-    return (file.indexOf('.') !== 0) && (file !== basename) && (file.slice(-3) === '.js');
-  })
-  .forEach(function(file) {
-    var model = sequelize['import'](path.join(__dirname, file));
-    db[model.name] = model;
-  });
+const sequelize = starting(config)
+const isModels = file => (file.indexOf('.') !== 0) && (file !== basename) && (file.slice(-3) === '.js')
```
Comment (Americas) [inline lib/assets/models/index.js]: `IsModel`

File: lib/assets/models/index.js [hunk 3]
```diff
@@ -1,36 +1,34 @@
...
+const modelImport = file => sequelize.import(path.join(__dirname, file))
```
Comment (Americas) [inline lib/assets/models/index.js]: `importModel`

File: lib/assets/models/index.js [hunk 4]
```diff
@@ -1,36 +1,34 @@
...
+    .map(model => modelImport(model))
+    .reduce((acc, model) => {
+      acc[model.name] = model
+      return acc
+    }, {})
 
-db.sequelize = sequelize;
-db.Sequelize = Sequelize;
+Object.keys(models)
+  .map(model => models[model].associate(models))
```
Comment (Americas) [inline lib/assets/models/index.js]: If model is created without an associate function, this throws an error.

Comment (danielsalles) [inline lib/assets/models/index.js]: I had some time now. I'll fix everything and commit.

**Comment by sushantdhiman** [comment]
Closing old PRs which require rebase (after v3), please let me know if you wish to work on this. In that case just rebase and ping me :)

**Pull request closed without merging**